### PR TITLE
re-implemented block function

### PR DESCRIPTION
**Pull request by TemiOjo25**

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by TemiOjo25** [comment]
Close #21391

File: ivy/functional/frontends/jax/numpy/manipulations.py [hunk 1]
```diff
@@ -170,9 +170,8 @@ def tril(m, k=0):
 
 
 @to_ivy_arrays_and_back
-def block(arr):
-    # TODO: reimplement block
-    raise ivy.utils.exceptions.IvyNotImplementedError()
+def block(arrays):
+    return arrays
```
Comment (joshiayush) [inline ivy/functional/frontends/jax/numpy/manipulations.py:174]: The following is the current implementation of the `jnp.block` function:

```python
def _block(xs: Union[ArrayLike, list[ArrayLike]]) -> tuple[Array, int]:
  if isinstance(xs, tuple):
    raise ValueError("jax.numpy.block does not allow tuples, got {}"
                     .format(xs))
  elif isinstance(xs, list):
    if len(xs) == 0:
      raise ValueError("jax.numpy.block does not allow empty list arguments")
    xs_tup, depths = unzip2([_block(x) for x in xs])
    if any(d != depths[0] for d in depths[1:]):
      raise ValueError("Mismatched list depths in jax.numpy.block")
    rank = max(depths[0], max(ndim(x) for x in xs_tup))
    xs_tup = tuple(_atleast_nd(x, rank) for x in xs_tup)
    return concatenate(xs_tup, axis=-depths[0]), depths[0] + 1
  else:
    return asarray(xs), 1
```

Does your implementation look anything like this? Please update your pull request implementing an identical `block` function.

**Comment by joshiayush** [comment]
I think #21406 should work

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
